### PR TITLE
[WIP] Admin Panel Rebuild

### PR DESCRIPTION
**Pull request by satyamurti**

### CODE REVIEW

File: app/src/main/java/org/pragyan/dalal18/utils/Constants.java [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
     // TODO (Release): Change server IP and Port
 
-    public static final String HOST = "dalal.pragyan.org";
+    public static final String HOST = "192.168.1.103";
```
Comment (avikumar15) [inline app/src/main/java/org/pragyan/dalal18/utils/Constants.java:7]: Change this back to previous url

File: app/src/main/java/org/pragyan/dalal18/fragment/adminPanel/AdminPanelUserSpecificFragment.kt [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+import org.pragyan.dalal18.databinding.FragmentAdminpanelStocksBinding
+import org.pragyan.dalal18.databinding.FragmentAdminpanelUserspecificBinding
+import org.pragyan.dalal18.databinding.FragmentSecretBinding
+import org.pragyan.dalal18.utils.ConnectionUtils
+import org.pragyan.dalal18.utils.Constants
+import org.pragyan.dalal18.utils.hideKeyboard
+import org.pragyan.dalal18.utils.viewLifecycle
+import javax.inject.Inject
+
+@Suppress("PLUGIN_WARNING")
```
Comment (avikumar15) [inline app/src/main/java/org/pragyan/dalal18/fragment/adminPanel/AdminPanelUserSpecificFragment.kt:31]: Where does this warning come?

File: app/src/main/java/org/pragyan/dalal18/fragment/adminPanel/AdminPanelUserSpecificFragment.kt [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        DaggerDalalStreetApplicationComponent.builder().contextModule(ContextModule(context!!)).build().inject(this)
+
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        binding.apply {
+
+        }
```
Comment (avikumar15) [inline app/src/main/java/org/pragyan/dalal18/fragment/adminPanel/AdminPanelUserSpecificFragment.kt:55]: Remove this

Comment (avikumar15) [reply app/src/main/java/org/pragyan/dalal18/fragment/adminPanel/AdminPanelUserSpecificFragment.kt:55]: The entire function, if not in use,

File: app/src/main/java/org/pragyan/dalal18/fragment/CompanyFragment.kt [hunk 4]
```diff
@@ -65,17 +65,18 @@ class CompanyFragment : Fragment(), CompanyRecyclerAdapter.OnCompanyClickListene
         DaggerDalalStreetApplicationComponent.builder().contextModule(ContextModule(context!!)).build().inject(this)
         updateValues()
         binding.apply {
-            portfolioRecyclerView.setHasFixedSize(false)
-            portfolioRecyclerView.adapter = adapter
-            portfolioRecyclerView.layoutManager = LinearLayoutManager(context)
+//            portfolioRecyclerView.setHasFixedSize(false)
+//            portfolioRecyclerView.adapter = adapter
+//            portfolioRecyclerView.layoutManager = LinearLayoutManager(context)
```
Comment (avikumar15) [inline app/src/main/java/org/pragyan/dalal18/fragment/CompanyFragment.kt:70]: Remove comments if not in use

File: app/src/main/java/org/pragyan/dalal18/fragment/adminPanel/AdminPanelDailyMarketFragment.kt [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+        super.onViewCreated(view, savedInstanceState)
+        binding.apply {
+            openMarketButton.setOnClickListener { openMarket() }
+            closeMarketButton.setOnClickListener { closeMarket() }
+            updateEndOFDayValue.setOnClickListener { updateEndOfDayValues() }
+            addDailyChallengeButton.setOnClickListener { addDailyChallenge() }
+            openDailyChallengeButton.setOnClickListener { openDailyChallenge() }
+            closeDailyChallengeButton.setOnClickListener { closeDailyChallenge() }
+
+
```
Comment (ranaaditya) [inline app/src/main/java/org/pragyan/dalal18/fragment/adminPanel/AdminPanelDailyMarketFragment.kt:61]: remove these extra lines !

File: app/src/main/java/org/pragyan/dalal18/fragment/adminPanel/AdminPanelUserSpecificFragment.kt [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+                ?: throw Exception("Invalid activity")
+        DaggerDalalStreetApplicationComponent.builder().contextModule(ContextModule(context!!)).build().inject(this)
+
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        binding.apply {
+
```
Comment (ranaaditya) [inline app/src/main/java/org/pragyan/dalal18/fragment/adminPanel/AdminPanelUserSpecificFragment.kt:54]: why  blank binding.apply{} exists here ?

either remove it completely or add it in comments !

File: app/src/main/java/org/pragyan/dalal18/fragment/adminPanel/AdminPanelUserSpecificFragment.kt [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+import javax.inject.Inject
+
+@Suppress("PLUGIN_WARNING")
+class AdminPanelUserSpecificFragment : Fragment() {
+    private var binding by viewLifecycle<FragmentAdminpanelUserspecificBinding>()
+    @Inject
+    lateinit var actionServiceBlockingStub: DalalActionServiceGrpc.DalalActionServiceBlockingStub
+
+    private lateinit var model: DalalViewModel
+    private var message = ""
```
Comment (ranaaditya) [inline app/src/main/java/org/pragyan/dalal18/fragment/adminPanel/AdminPanelUserSpecificFragment.kt:38]: can't we make it **lateinit** ?

File: app/src/main/java/org/pragyan/dalal18/utils/Constants.java (hunk 1 above)
Comment (ranaaditya) [inline app/src/main/java/org/pragyan/dalal18/utils/Constants.java:7]: revert it back to original configurations !

File: app/src/main/res/layout/fragment_adminpanel_dailymarket.xml [hunk 8]
```diff
@@ -0,0 +1,307 @@
...
+                android:text="Close Market"
+                android:textColor="@color/neon_yellow"
+                app:layout_constraintEnd_toEndOf="@id/openMarketButton"
+                app:layout_constraintHorizontal_bias="0.0"
+                app:layout_constraintStart_toStartOf="@id/openMarketButton"
+                app:layout_constraintTop_toBottomOf="@id/openMarketButton" />
+
+
+            <View
+                android:id="@+id/view3"
```
Comment (ranaaditya) [inline app/src/main/res/layout/fragment_adminpanel_dailymarket.xml:61]: use proper names for view naming !

File: app/src/main/res/layout/fragment_adminpanel_dailymarket.xml [hunk 9]
```diff
@@ -0,0 +1,307 @@
...
+                android:layout_width="match_parent"
+                android:layout_height="2dp"
+                android:layout_margin="5dp"
+                android:background="@color/black"
+                app:layout_constraintTop_toBottomOf="@+id/closeMarketButton"
+                tools:layout_editor_absoluteX="8dp" />
+
+            <TextView
+                android:id="@+id/stockIdEditText2"
```
Comment (ranaaditya) [inline app/src/main/res/layout/fragment_adminpanel_dailymarket.xml:70]: name it properly !

File: app/src/main/res/layout/fragment_adminpanel_dailymarket.xml [hunk 10]
```diff
@@ -0,0 +1,307 @@
...
+                android:textColorHint="@color/neutral_font_color"
+                app:layout_constraintBottom_toBottomOf="@+id/updateEndOFDayValue"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toTopOf="@+id/updateEndOFDayValue" />
+
+            <Button
+                android:id="@+id/updateEndOFDayValue"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_marginTop="16dp"
```
Comment (ranaaditya) [inline app/src/main/res/layout/fragment_adminpanel_dailymarket.xml:86]: remove hard coded paddings and margins everywhere in this xml layout file !

use dimens.xml for this

File: app/src/main/res/layout/fragment_adminpanel_dailymarket.xml [hunk 11]
```diff
@@ -0,0 +1,307 @@
...
+                android:background="@drawable/black_white_rounded_button"
+                android:padding="10dp"
+                android:text="Close DailyChallenge"
+                android:textColor="@color/neon_yellow"
+                app:layout_constraintEnd_toEndOf="@+id/openDailyChallengeButton"
+                app:layout_constraintHorizontal_bias="0.166"
+                app:layout_constraintStart_toStartOf="@+id/openDailyChallengeButton"
+                app:layout_constraintTop_toBottomOf="@+id/openDailyChallengeButton" />
+            <View
+                android:id="@+id/view4"
```
Comment (ranaaditya) [inline app/src/main/res/layout/fragment_adminpanel_dailymarket.xml:255]: bad way for naming a layout !

This may looks fine for now but not a good practice to handle OpenSource projects !

Please rename all views to some meaning names :)

File: app/src/main/res/layout/fragment_adminpanel_userspecific.xml [hunk 12]
```diff
@@ -0,0 +1,234 @@
...
+            android:padding="4dp"
+            android:text="Block User"
+            android:textColor="@color/red_piechart"
+            app:layout_constraintEnd_toEndOf="@+id/blockuserEditText"
+            app:layout_constraintHorizontal_bias="0.15"
+            app:layout_constraintStart_toStartOf="@+id/blockuserEditText"
+            app:layout_constraintTop_toBottomOf="@+id/blockuserEditText" />
+
+        <View
+            android:id="@+id/view3"
```
Comment (ranaaditya) [inline app/src/main/res/layout/fragment_adminpanel_userspecific.xml:62]: rename it also !

File: app/src/main/res/layout/fragment_adminpanel_userspecific.xml [hunk 13]
```diff
@@ -0,0 +1,234 @@
...
+            android:background="@drawable/black_white_rounded_button"
+            android:padding="4dp"
+            android:text="Unblock User"
+            android:textColor="@color/neon_green"
+            app:layout_constraintEnd_toEndOf="@+id/unblockuserEditText"
+            app:layout_constraintStart_toStartOf="@+id/unblockuserEditText"
+            app:layout_constraintTop_toBottomOf="@+id/unblockuserEditText" />
+
+        <View
+            android:id="@+id/view5"
```
Comment (ranaaditya) [inline app/src/main/res/layout/fragment_adminpanel_userspecific.xml:113]: rename it !

File: app/src/main/res/layout/fragment_adminpanel_userspecific.xml [hunk 14]
```diff
@@ -0,0 +1,234 @@
...
+            android:padding="4dp"
+            android:text="Unblock All User"
+            android:textColor="@color/neon_green"
+            app:layout_constraintEnd_toEndOf="@+id/unblockUserButton"
+            app:layout_constraintHorizontal_bias="0.0"
+            app:layout_constraintStart_toStartOf="@+id/unblockUserButton"
+            app:layout_constraintTop_toBottomOf="@+id/view5" />
+
+        <View
+            android:id="@+id/view4"
```
Comment (ranaaditya) [inline app/src/main/res/layout/fragment_adminpanel_userspecific.xml:150]: rename it !

File: app/src/main/res/layout/fragment_adminpanel_userspecific.xml [hunk 15]
```diff
@@ -0,0 +1,234 @@
...
+            app:layout_constraintStart_toEndOf="@+id/unblockUserTextview"
+            app:layout_constraintTop_toBottomOf="@+id/view3" />
+
+        <Button
+            android:id="@+id/unblockUserButton"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="12dp"
+            android:background="@drawable/black_white_rounded_button"
+            android:padding="4dp"
```
Comment (ranaaditya) [inline app/src/main/res/layout/fragment_adminpanel_userspecific.xml:105]: remove hard coded paddings and margins everywhere from this layout file.

**Review by ranaaditya — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/pragyan/dalal18/fragment/CompanyFragment.kt (hunk 4 above)
Comment (avikumar15) [reply app/src/main/java/org/pragyan/dalal18/fragment/CompanyFragment.kt:70]: @satyamurti

File: app/src/main/java/org/pragyan/dalal18/fragment/adminPanel/AdminPanelUserSpecificFragment.kt [hunk 16]
```diff
@@ -0,0 +1,166 @@
...
+                                .setTransactionType(askUserSwitch.isChecked)
+                                .build())
+                        resp = response.listList
+//                        resp.forEach{
+//                            context?.longToast("User Email : ${it.email} \n" +
+//                                    "User Id : ${it.id} \n " +
+//                                    "Position : ${it.position} \n" +
+//                                    "StockSum : ${it.stockSum} \n " +
+//                                    "TransactionCount : ${it.transactionCount}")
+//                        }
```
Comment (avikumar15) [inline app/src/main/java/org/pragyan/dalal18/fragment/adminPanel/AdminPanelUserSpecificFragment.kt:152]: Remove this as well

File: app/src/main/java/org/pragyan/dalal18/fragment/adminPanel/AdminPanelUserSpecificFragment.kt [hunk 17]
```diff
@@ -0,0 +1,166 @@
...
+                        message = response.statusMessage
+                    }
+                }
+            }
+        }
+        d("SPD", resp.toString())
```
Comment (avikumar15) [inline app/src/main/java/org/pragyan/dalal18/fragment/adminPanel/AdminPanelUserSpecificFragment.kt:158]: Remove this log

**Comment by ranaaditya** [comment]
@satyamurti  are you still getting the empty list ?

File: app/src/main/res/values/strings.xml [hunk 18]
```diff
@@ -487,4 +487,12 @@
...
     <string name="reward">Rewards</string>
     <string name="challenge">Challenges</string>
+    <string-array name="dailyChallengesTypes">
+        <item>CASH</item>
+        <item>NetWorth</item>
+        <item>SpecificStock</item>
+        <item>StockWorth</item>
+    </string-array>
+    <!-- TODO: Remove or change this placeholder text -->
+    <string name="hello_blank_fragment">Hello blank fragment</string>
```
Comment (ranaaditya) [inline app/src/main/res/values/strings.xml:497]: remove this auto-generated  comment and string.

File: app/src/main/res/layout/inspect_user_list_item.xml [hunk 19]
```diff
@@ -0,0 +1,82 @@
...
+            android:text="UserId : 01"
+            android:textColor="@color/neutral_font_color"
+            android:textSize="22sp"
+            android:textStyle="bold"
+            app:layout_constraintBottom_toTopOf="@id/email_textView"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent" />
+
+        <TextView
+            android:id="@+id/email_textView"
```
Comment (ranaaditya) [inline app/src/main/res/layout/inspect_user_list_item.xml:32]: maintain consistency in naming UI variables.

Sometime you have used snake case and sometimes camel cases, remove this difference for this file to have smooth reading and reviewing experience.
@satyamurti

File: app/src/main/res/layout/fragment_adminpanel_dailymarket.xml [hunk 20]
```diff
@@ -0,0 +1,307 @@
...
+            android:padding="10dp"
+            android:text="Set Market Day"
+            android:textColor="@color/neon_yellow"
+            app:layout_constraintEnd_toEndOf="@+id/setMarketdayEditText"
+            app:layout_constraintHorizontal_bias="0.0"
+            app:layout_constraintStart_toStartOf="@+id/setMarketdayEditText"
+            app:layout_constraintTop_toBottomOf="@+id/setMarketdayEditText" />
+
+        <TextView
+                android:id="@+id/setMarketDayTExtView"
```
Comment (ranaaditya) [inline app/src/main/res/layout/fragment_adminpanel_dailymarket.xml:293]: name it properly @satyamurti

File: app/src/main/res/layout/fragment_adminpanel_dailymarket.xml [hunk 21]
```diff
@@ -0,0 +1,307 @@
...
+                android:inputType="number"
+                android:text="Daily LeaderBoard"
+                android:textColor="@color/neutral_font_color"
+                android:textColorHint="@color/neutral_font_color"
+                app:layout_constraintBottom_toBottomOf="@+id/updateEndOFDayValue"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toTopOf="@+id/updateEndOFDayValue" />
+
+            <Button
+                android:id="@+id/updateEndOFDayValue"
```
Comment (ranaaditya) [inline app/src/main/res/layout/fragment_adminpanel_dailymarket.xml:83]: name it properly to **updateEndOfDayValue** @satyamurti

File: app/src/main/res/layout/fragment_adminpanel_dailymarket.xml [hunk 22]
```diff
@@ -0,0 +1,307 @@
...
+                android:text="Close Market"
+                android:textColor="@color/neon_yellow"
+                app:layout_constraintEnd_toEndOf="@id/openMarketButton"
+                app:layout_constraintHorizontal_bias="0.0"
+                app:layout_constraintStart_toStartOf="@id/openMarketButton"
+                app:layout_constraintTop_toBottomOf="@id/openMarketButton" />
+
+
+            <View
+                android:id="@+id/viewForDivingSections3"
```
Comment (ranaaditya) [inline app/src/main/res/layout/fragment_adminpanel_dailymarket.xml:61]: please again check all the variable names for your Pr @satyamurti

**Review by ranaaditya — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/pragyan/dalal18/utils/Constants.java [hunk 23]
```diff
@@ -4,7 +4,7 @@
 
     // TODO (Release): Change server IP and Port
 
-    public static final String HOST = "venmei.com";
+    public static final String HOST = "venmei.comz";
```
Comment (ranaaditya) [inline app/src/main/java/org/pragyan/dalal18/utils/Constants.java:7]: revert this change !

File: app/src/main/res/layout/fragment_adminpanel_stocks.xml [hunk 24]
```diff
@@ -0,0 +1,267 @@
...
+            android:background="@drawable/black_white_rounded_button"
+            android:text="Send News"
+            android:textColor="@color/neon_blue"
+            app:layout_constraintBottom_toBottomOf="@id/newsDescriptionEditText"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toEndOf="@+id/headlinesEditText"
+            app:layout_constraintTop_toTopOf="@+id/isGlobalNewsSwitch" />
+
+        <View
+            android:id="@+id/viewForDivingSections2"
```
Comment (ranaaditya) [inline app/src/main/res/layout/fragment_adminpanel_stocks.xml:254]: remove suffix **2** from variable name !

File: app/src/main/res/layout/fragment_adminpanel_userspecific.xml [hunk 25]
```diff
@@ -0,0 +1,325 @@
...
+            android:text="Unblock User"
+            android:textColor="@color/neon_green"
+            app:layout_constraintEnd_toEndOf="@+id/unblockUserEditText"
+            app:layout_constraintStart_toStartOf="@+id/unblockUserEditText"
+            app:layout_constraintTop_toBottomOf="@+id/unblockUserEditText" />
+
+        <View
+            android:id="@+id/view5"
```
Comment (ranaaditya) [inline app/src/main/res/layout/fragment_adminpanel_userspecific.xml:113]: rename it properly !

**Review by ranaaditya — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/pragyan/dalal18/fragment/SecretFragment.kt [hunk 26]
```diff
@@ -51,164 +46,160 @@ class SecretFragment : Fragment() {
...
+//                                .setImageUrl(newsImageUrlEditText.text.toString())
+//                                .setIsGlobal(isGlobalNewsSwitch.isChecked)
+//                                .build())
+//                        message = response.statusMessage
+//                    }
+//                }
+//            }
+//        }
+//        context?.toast(message)
+//    }
```
Comment (avikumar15) [inline app/src/main/java/org/pragyan/dalal18/fragment/SecretFragment.kt:202]: Remove this

**Review by ranaaditya — APPROVED** [review]

**Review by ranaaditya — APPROVED** [review]

**Review by avikumar15 — APPROVED** [review]
